staker/block_validator.go

```diff
@@ -796,6 +843,14 @@ func (v *BlockValidator) iterativeValidationPrint(ctx context.Context) time.Dura
 	}
 	log.Info("validated execution", "messageCount", printedCount, "globalstate", validated.GlobalState, "WasmRoots", validated.WasmRoots)
 	v.lastValidInfoPrinted = validated
+	revalidationConfig := v.config().Dangerous.Revalidation
+	if revalidationConfig.EndBlock > 0 && validated.GlobalState.Batch >= revalidationConfig.EndBlock {
+		if revalidationConfig.QuitAfterRevalidation {
+			v.fatalErr <- fmt.Errorf("revalidation done from %d to %d", revalidationConfig.StartBlock, revalidationConfig.EndBlock)
```

Comment:
Semantically this is not an error. 
The approach of having a separate channel for non error reasons to shut down a node seems more appropriate.
As example, if we use fatalErr channel here then the node process will exit with status 1, instead of 0.
This can make the life of node operators less straightforward, when creating automation processes that decide if a node should be automatically restarted, or inspected, etc.

Reply:
Updated to send nil error and add a log.